pravega-client-examples/src/main/java/io/pravega/example/secure/SecureBatchReader.java

```
@@ -0,0 +1,319 @@
+           e.printStackTrace();
+        }
+        LOG.info("Started segment logger");
+    }
+
+    private void logEventCounts() {
+        try (BatchClientFactory batchClient = BatchClientFactory.withScope(scope, clientConfig)) {
+             System.out.println("Done creating batchClient for " + scope + "/" + stream);
+             ArrayList<SegmentRange> ranges = Lists.newArrayList(batchClient.getSegments(Stream.of(scope, stream), StreamCut.UNBOUNDED, StreamCut.UNBOUNDED).getIterator());
+             readFromRanges(batchClient, ranges);
+             System.out.println("Done reading ranges of size: " + String.valueOf(ranges.size()));
+        } catch (Exception e) {
+            e.printStackTrace();
+            LOG.error("Exception:", e);
+        }
+    }
+
+    private int readFromRanges(BatchClientFactory batchClient, List<SegmentRange> ranges) {
+        LOG.info("Number of ranges: {}", ranges.size());
+        List<CompletableFuture<Integer>> eventCounts = new ArrayList<>();
+        JavaSerializer<String> serializer = new JavaSerializer<String>();
+        eventCounts = ranges
+                .parallelStream()
+                .map(range -> CompletableFuture.supplyAsync(() -> batchClient.readSegment(range, serializer))
+                        .thenApplyAsync(segmentIterator -> {
+                            int numEvents = 0;
+                            try {
+                                String id = String.valueOf(Thread.currentThread().getId());
+                                while (segmentIterator.hasNext()) {
+                                    String event = segmentIterator.next();
+                                    System.out.println("Done reading event by thread " + id + ": " + event);
+                                    numEvents++;
+                                }
+                            } catch (Exception e) {
+                                e.printStackTrace();
```

Let the exception bubble up.

---

Yes Andre